staging/src/k8s.io/sample-apiserver/docs/minikube-walkthrough.md

```diff
@@ -86,18 +97,114 @@ kubectl create -f artifacts/example/service.yaml -n wardle
 kubectl create -f artifacts/example/apiservice.yaml
 ```
 
-## Test that your setup has worked
+You can skip to [test your setup](#test-that-your-setup-has-worked)
+
+## Developing on your machine
+
+This section will focus on how to develop and run the sample apisever on your machine and use minikube to host `etcd`.
 
-You should now be able to create the resource type ```Flunder``` which is the resource type registered by the sample apiserver.
+### Deploying etcd and setting services
 
+```bash
+
+# you'll need the IP address of your machine, otherwise minikube's apiserver cannot talk to your sample-apiserver.
+# On Linux:
+hostname -I
+# On OSX:
+ipconfig getifaddr en0
+
+MACHINE_IP=<YOUR_MACHINE_IP>
+MINIKUBE_IP=$(minikube ip)
+
+# create the namespace to run the apiserver in
+kubectl create ns wardle
+
+# create the etcd
+cat <<EOF | kubectl create -f -
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: wardle-etcd
+  namespace: wardle
+  labels:
+    etcd: "true"
+spec:
+  replicas: 1
+  selector:
+    etcd: "true"
+  template:
+    metadata:
+      labels:
+        etcd: "true"
+    spec:
+      containers:
+      - image: quay.io/coreos/etcd:v3.1.10
+        name: etcd
+        command:
+        - etcd
+        - -advertise-client-urls=http://${MINIKUBE_IP}:2379
```

Comment:
do we really need this copy of the file here? Can't we describe how to replace the IP in the provided file?

Reply:
You are referring to sample apisever + etcd replicationcontroller [manifest file](https://github.com/kubernetes/kubernetes/blob/7652c252d4ce2aea1563853bef0438769c5781fd/staging/src/k8s.io/sample-apiserver/artifacts/example/rc.yaml) or copy pasting into the terminal?

Reply:
The former.